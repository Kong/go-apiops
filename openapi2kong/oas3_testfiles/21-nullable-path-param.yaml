openapi: 3.0.0
info:
  title: OData style API with nullable path params
  version: 0.0.1
servers:
  - url: http://backend.com/path
paths:
  /custom/{customId}:
    parameters:
      - name: customId
        in: path
        required: true
        description: Custom ID
        schema:
          type: string
          maxLength: 4
    get:
      summary: Read the custom information
      description: Read the custom information of the API
      tags:
        - Metadata
      responses:
        "200":
          description: Retrieved metadata
        4XX:
          description: all 4XX errors
      operationId: customdata
  ? "/CustomWarehouseData(Warehouse='{Warehouse}',Product='{Product}',Location='{Location}')"
  : parameters:
      - name: Warehouse
        in: path
        required: true
        description: ID of warehouse
        schema:
          type: string
          maxLength: 4
          nullable: true
      - name: Product
        in: path
        required: true
        description: Product Number
        schema:
          type: string
          maxLength: 40
      - name: Location
        in: path
        required: true
        description: Product Number
        schema:
          type: string
          maxLength: 40
    get:
      summary: List entities from Warehouse based on product ID, location and warehouse ID
      parameters:
        - name: Location #overrides the param defined in path level
          in: path
          required: true
          description: Product Number
          schema:
            type: string
            maxLength: 40
            nullable: true
      tags:
        - WarehouseProductData
      responses:
        "200":
          description: Retrieved product entities
        4XX:
          description: all 4XX errors
      operationId: customproductdata