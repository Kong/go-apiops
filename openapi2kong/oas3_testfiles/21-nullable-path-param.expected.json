{
    "_format_version": "3.0",
    "services": [
      {
        "host": "backend.com",
        "id": "aac33fce-7ade-5b69-9249-97bea965698e",
        "name": "odata-style-api-with-nullable-path-params",
        "path": "/path",
        "plugins": [],
        "port": 80,
        "protocol": "http",
        "routes": [
          {
            "id": "649b2718-47a3-57ec-aec0-b8a4f1bf010d",
            "methods": [
              "GET"
            ],
            "name": "odata-style-api-with-nullable-path-params_customproductdata",
            "paths": [
              "~/CustomWarehouseData\\(Warehouse='(?\u003cwarehouse\u003e[^#?/]*)',Product='(?\u003cproduct\u003e[^#?/]+)'\\)$"
            ],
            "plugins": [],
            "regex_priority": 100,
            "strip_path": false,
            "tags": [
              "OAS3_import",
              "OAS3file_21-nullable-path-param.yaml"
            ]
          },
          {
            "id": "dd39d9e0-196d-509d-b141-13e77ce94f81",
            "methods": [
              "GET"
            ],
            "name": "odata-style-api-with-nullable-path-params_customdata",
            "paths": [
              "~/custom/(?\u003ccustomid\u003e[^#?/]+)$"
            ],
            "plugins": [],
            "regex_priority": 100,
            "strip_path": false,
            "tags": [
              "OAS3_import",
              "OAS3file_21-nullable-path-param.yaml"
            ]
          }
        ],
        "tags": [
          "OAS3_import",
          "OAS3file_21-nullable-path-param.yaml"
        ]
      }
    ],
    "upstreams": []
  }